File: summer-school/MPI/Exercise_01/solution_01.c
#include <stdlib.h>
#include <stdio.h>
#include <mpi.h>

int main(int argc, char* argv[]){

  int me, nprocs;

  /* Initialize MPI environment */
  MPI_Init(&argc, &argv) ;
    
  /* Get the size of the MPI_COMM_WORLD communicator */
  MPI_Comm_size(MPI_COMM_WORLD, &nprocs) ;
   
  /* Get my rank... */
  MPI_Comm_rank(MPI_COMM_WORLD, &me) ;
    
  /* ...and print it. */
  printf("\tHello, I am task %d of %d.\n", me, nprocs);
  /* Finalize MPI environment */
  MPI_Finalize() ;

  return 0;
}

File: summer-school/MPI/Exercise_01/solution_01.f90
program hello

  !$ Use MPI module in Fortran 90
  use mpi
  implicit none

  integer ierr,me,nprocs

  !$ Initialize MPI environment
  call MPI_INIT(ierr)
  
  !$ Get the size of the MPI_COMM_WORLD communicator
  call MPI_COMM_SIZE(MPI_COMM_WORLD, nprocs, ierr)
  
  !$ Get my rank... (zero-based even if we are in Fortran)
  call MPI_COMM_RANK(MPI_COMM_WORLD, me, ierr)
  
  !$ ...and print it.
  print *,"Hello, I am task ",me," of ",nprocs,"."
  
  !$  Finalize MPI environment
  call MPI_FINALIZE(ierr)

end program hello
